src/pudl/workspace/datastore.py

```diff
@@ -1,715 +1,537 @@
+
+        Return:
+            str: a path
+        """
+        doi_dirname = re.sub("/", "-", self.doi(dataset))
+        directory = self.pudl_in / "data" / dataset / doi_dirname
+
+        if filename is None:
+            return directory
+
+        return directory / filename
+
+    def save_datapackage_json(self, dataset, dpkg):
+        """
+        Save a datapackage.json file.  Overwrite any previous version.
+
+        Args:
+            dataset (str): name of the dataset, as available in DOI
+            dpkg (dict): dict matching frictionless datapackage spec
+
+        Returns:
+            Path of the saved datapackage
+        """
+        path = self.local_path(dataset, filename="datapackage.json")
+        path.parent.mkdir(parents=True, exist_ok=True)
+        text = json.dumps(dpkg, sort_keys=True, indent=4)
+
+        with path.open("w") as f:
+            f.write(text)
+            self.logger.debug("%s saved.", path)
+        self._validate_datapackage(dataset)
+
+        return path
+
+    # Datapackage metadata
+
+    def remote_datapackage_json(self, doi):
+        """
+        Produce the contents of a remote datapackage.json.
+
+        Args:
+            doi: the DOI
+
+        Returns:
+            dict representation of the datapackage.json file as available on
+            Zenodo, or raises an error
+        """
+        dpkg_url = self.doi_to_url(doi)
+        response = self.http.get(
+            dpkg_url, params={"access_token": self.token}, timeout=self.timeout)
+
+        if response.status_code > 299:
+            msg = "Failed to retrieve %s" % dpkg_url
+            self.logger.error(msg)
+            raise ValueError(msg)
+
+        jsr = response.json()
+        files = {x["filename"]: x for x in jsr["files"]}
+
+        response = self.http.get(
+            files["datapackage.json"]["links"]["download"],
+            params={"access_token": self.token},
+            timeout=self.timeout)
+
+        if response.status_code > 299:
+            msg = "Failed to retrieve datapackage for %s: %s" % (
+                doi, response.text)
+            self.logger.error(msg)
+            raise ValueError(msg)
+
+        return json.loads(response.text)
+
+    def datapackage_json(self, dataset, force_download=False):
+        """
+        Produce the contents of datapackage.json, cache as necessary.
+
+        Args:
+            dataset (str): name of the dataset, must be available in the DOIS
+
+        Return:
+            dict: representation of the datapackage.json
+
+        """
+        path = self.local_path(dataset, filename="datapackage.json")
+
+        if force_download or not path.exists():
+            doi = self.doi(dataset)
+            dpkg = self.remote_datapackage_json(doi)
+            self.save_datapackage_json(dataset, dpkg)
+
+        with path.open("r") as f:
+            return yaml.safe_load(f)
+
+    # Remote resource retrieval
+
+    def is_remote(self, resource):
```

Comment:
Codacy found an issue: [Method could be a function](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=6263807)